### PR TITLE
[cluster-autoscaler] Add leader election to cluster autoscaler

### PR DESCRIPTION
**Pull request by fgrzadkowski**
Please note that this changes how we can pass flags to CA. In particular `-v` is incorrect and should be `--v`. Command lines in kubernetes scripts will have to be updated when releasing CA.

https://github.com/kubernetes/kubernetes/issues/21124

@jszczepkowski

### CODE REVIEW

**Comment by fgrzadkowski** [comment]
@mwielgus Can you please take a look?

File: cluster-autoscaler/cluster_autoscaler.go [hunk 1]
```diff
@@ -72,31 +75,29 @@ var (
...
+}
+
+func createEventRecorder(kubeClient *kube_client.Client) kube_record.EventRecorder {
+	eventBroadcaster := kube_record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(kubeClient.Events(""))
+	return eventBroadcaster.NewRecorder(kube_api.EventSource{Component: "cluster-autoscaler"})
+}
+
+func run(_ <-chan struct{}) {
```
Comment (mwielgus) [inline cluster-autoscaler/cluster_autoscaler.go]: what is the purpose of the argument

**Comment by mwielgus** [comment]
Just add comment to run. Otherwise LGTM

**Comment by fgrzadkowski** [comment]
Added comment.

**Pull request merged**